### PR TITLE
XSD Schema for XML rules

### PR DESCRIPTION
**Pull request by mbriskar**
Not easily made extensible because of http://en.wikipedia.org/wiki/Unique_Particle_Attribution (only for XSD 1.1)

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Merged build triggered.

**Comment by windupgithubbot1** [comment]
Merged build started.

**Comment by windupgithubbot1** [comment]
Merged build finished. Test FAILed.

**Comment by windupgithubbot1** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://10.3.9.192//job/windup-pr-builder/437/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Merged build triggered.

File: config-xml/rule-schema.xsd [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+						<xs:element name="iteration" type="iteration-all">
+						</xs:element>
+					</xs:sequence>
+				</xs:complexType>
+			</xs:element>
+		</xs:sequence>
+	</xs:complexType>
+
+
+	<xs:element name="ruleset">
```
Comment (mbriskar) [inline config-xml/rule-schema.xsd]: This is the main core of XSD referencing other "element" and "type" definitions listed above. Some of them are not used yet (rules-xml, rules-java, rules-project with dependencies)

**Comment by windupgithubbot1** [comment]
Merged build triggered.

**Comment by windupgithubbot1** [comment]
Merged build started.

**Comment by windupgithubbot1** [comment]
Merged build finished. Test FAILed.

**Comment by windupgithubbot1** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://10.3.9.192//job/windup-pr-builder/438/
Test FAILed.

File: rules-xml/src/main/java/org/jboss/windup/rules/apps/xml/confighandler/XmlFileHandler.java [hunk 2]
```diff
@@ -51,7 +53,7 @@ public XmlFile processElement(ParserContext handlerManager, Element element)
         {
             throw new WindupException("Error, 'xmlfile' element must have a non-empty 'matches' or public-id attribute");
         }
-        String inFile = $(element).attr("in-file");
```
Comment (mbriskar) [inline rules-xml/src/main/java/org/jboss/windup/rules/apps/xml/confighandler/XmlFileHandler.java]: in is used in the JavaClass also and all of our xml rules were using "in" and not "in-file", so I switched it here

**Comment by windupgithubbot1** [comment]
Merged build triggered.

File: rules-xml/src/main/java/org/jboss/windup/rules/apps/xml/confighandler/XmlFileHandler.java (hunk 2 above)
Comment (lincolnthree) [inline rules-xml/src/main/java/org/jboss/windup/rules/apps/xml/confighandler/XmlFileHandler.java]: We should update the other files, not this :)

Comment (lincolnthree) [inline rules-xml/src/main/java/org/jboss/windup/rules/apps/xml/confighandler/XmlFileHandler.java]: Sorry, I got it backwards. You did the right thing!

**Comment by windupgithubbot1** [comment]
Merged build triggered.

**Comment by windupgithubbot1** [comment]
Merged build started.

**Comment by windupgithubbot1** [comment]
Merged build finished. Test FAILed.

**Comment by windupgithubbot1** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://10.3.9.192//job/windup-pr-builder/439/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Merged build triggered.

**Comment by jsight** [comment]
retest

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://10.3.9.192//job/windup-pr-builder/440/
Test FAILed.

**Comment by jsight** [comment]
retest

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://10.3.9.192//job/windup-pr-builder/442/
Test FAILed.

**Comment by lincolnthree** [comment]
retest

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://10.3.9.192//job/windup-pr-builder/459/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules.apps:rules-java-ee' /><a href='https://10.3.9.192/job/windup-pr-builder/459/org.jboss.windup.rules.apps$rules-java-ee/testReport'>org.jboss.windup.rules.apps:rules-java-ee</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://10.3.9.192/job/windup-pr-builder/459/org.jboss.windup.rules.apps$rules-java-ee/testReport/org.jboss.windup.rules.apps.javaee.tests/XmlJonasConfigTest/testHintsAndClassificationOperation/'><strong>org.jboss.windup.rules.apps.javaee.tests.XmlJonasConfigTest.testHintsAndClassificationOperation</strong></a></li></ul>
Test FAILed.

**Comment by mbriskar** [comment]
retest this please

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://10.3.9.192//job/windup-pr-builder/461/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://10.3.9.192//job/windup-pr-builder/462/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://10.3.9.192//job/windup-pr-builder/463/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://10.3.9.192//job/windup-pr-builder/465/
Test PASSed.

**Comment by lincolnthree** [comment]
@mbriskar Please reference the JIRA ID in the commit message: WINDUP-481

Then we can merge. Thanks.

**Comment by lincolnthree** [comment]
Merged.

**Pull request closed without merging**